bin/do_metal_xdmat.py

```diff
@@ -136,8 +136,47 @@ def calc_metal_xdmat(abs_igm,p):
                                 args.z_evol_obj, args.z_ref,cosmo)
     xcf.objs = objs
 
+    ### Remove pixels if too far from objects
+    zmin_obj = None
+    zmax_obj = None
+    for pix in xcf.objs:
+        for q in xcf.objs[pix]:
+            if zmin_obj is None: zmin_obj = q.zqso
+            if zmax_obj is None: zmax_obj = q.zqso
+            zmin_obj = min(zmin_obj,q.zqso)
+            zmax_obj = max(zmax_obj,q.zqso)
+    if ( zmin_obj is not None and ( (zmin_pix<zmin_obj) or (zmax_obj<zmax_pix)) ):
```

Comment:
zmin_obj can't be None at this point. Can you simplify the expression?

Reply:
Yes it can if there are no objetc in the catalog for the required zmin_pix and zmax_pix